### DISCUSSION TITLE
Make Storage::download accept absolute path to a file

### DISCUSSION THREAD

**Discussion by bokunodev** [laravel/framework#39866] [Ideas]
I'm using laravel with phpword on a shared hosting.
The issue is that php word create doc file in system temporary directory.
the solution is either make phpword to use custom temp directory or allow `Storage::download` to accept absolute path.
When i did 
```php
$tempDocFile = $templateProccessoor->save();
return Storage::download($tempDocFile);
```
it gave me this
![image](https://user-images.githubusercontent.com/54968427/144427933-ed7f7d7a-4c3d-4289-becb-68a1b54c2f0b.png)

it seems like `Storage::download` strip leading `/`. sure i can use `$tpp->saveAs($path)` instead,
but that means i have to configure cronjob and perform cleanup by myself instead of let the hosting provider do it for me.

**Comment by kn0x1c** [comment]
Same issue here, have to rebuild my models!
